pkg/frontend/fixetcd.go

```diff
@@ -492,141 +484,162 @@ func createPrivilegedServiceAccount(ctx context.Context, log *logrus.Entry, name
 //
 // If backups already exists the job is cowardly and refuses to overwrite them
 func backupEtcdData(ctx context.Context, log *logrus.Entry, cluster, node string, kubeActions adminactions.KubeActions) ([]byte, error) {
-	jobDataBackup := createBackupEtcdDataJob(cluster, node)
+	podDataBackup, err := createBackupEtcdDataPod(node)
 
-	log.Infof("Creating job %s", jobDataBackup.GetName())
-	err := kubeActions.KubeCreateOrUpdate(ctx, jobDataBackup)
 	if err != nil {
 		return []byte{}, err
 	}
-	log.Infof("Job %s has been created", jobDataBackup.GetName())
 
-	watcher, err := kubeActions.KubeWatch(ctx, jobDataBackup, "app")
+	log.Infof("Creating pod %s", podDataBackup.GetName())
+	err = kubeActions.KubeCreateOrUpdate(ctx, podDataBackup)
 	if err != nil {
 		return []byte{}, err
 	}
+	log.Infof("Pod %s has been created", podDataBackup.GetName())
 
-	containerLogs, err := waitForJobSucceed(ctx, log, watcher, jobDataBackup, kubeActions)
+	watcher, err := kubeActions.KubeWatch(ctx, podDataBackup, "app")
+	if err != nil {
+		return []byte{}, err
+	}
+	containerLogs, err := waitForPodSucceed(ctx, log, watcher, podDataBackup, kubeActions)
 	if err != nil {
 		return containerLogs, err
 	}
 
-	log.Infof("Deleting job %s now", jobDataBackup.GetName())
+	log.Infof("Deleting pod %s now", podDataBackup.GetName())
 	propPolicy := metav1.DeletePropagationBackground
-	return containerLogs, kubeActions.KubeDelete(ctx, "Job", namespaceEtcds, jobDataBackup.GetName(), true, &propPolicy)
+	return containerLogs, kubeActions.KubeDelete(ctx, "Pod", namespaceEtcds, podDataBackup.GetName(), true, &propPolicy)
 }
 
-func waitForJobSucceed(ctx context.Context, log *logrus.Entry, watcher watch.Interface, o *unstructured.Unstructured, k adminactions.KubeActions) ([]byte, error) {
+// Waits until a standalone pod succeeds or fails. There is no retry logic for failures.
+func waitForPodSucceed(ctx context.Context, log *logrus.Entry, watcher watch.Interface, o *unstructured.Unstructured, k adminactions.KubeActions) ([]byte, error) {
 	var waitErr error
+	var pod corev1.Pod
+	var originalPod corev1.Pod
 	log.Infof("Waiting for %s to reach %s phase", o.GetName(), corev1.PodSucceeded)
-	select {
-	case event := <-watcher.ResultChan():
-		pod := event.Object.(*corev1.Pod)
-
-		switch pod.Status.Phase {
-		case corev1.PodSucceeded:
-			log.Infof("Job %s completed with %s", pod.GetName(), pod.Status.Message)
-		case corev1.PodFailed:
-			log.Infof("Job %s reached phase %s with message: %s", pod.GetName(), pod.Status.Phase, pod.Status.Message)
-			waitErr = fmt.Errorf("pod %s event %s received with message %s", pod.Name, pod.Status.Phase, pod.Status.Message)
+outer:
```

Comment:
I mostly advise against using loop labels to break out of nested loops, as i find they make the code more confusing. But when using it in a `for{ select{...} }` block, i find them tolerable :) 

Alternatively, you could move this whole block to a separate function that just returns an error, as the block doesn't seem to have any other dependencies to what comes after it.

Reply:
I moved the loop into a separate function (I actually renamed the existing function so the name honors what it does). You can check this change in the latest commit.